backend/test_observer/controllers/promoter/promoter.py

```diff
@@ -119,14 +120,17 @@ def run_snap_promoter(session: Session, artefact: Artefact) -> None:
     :session: DB connection session
     :artefact_build: an ArtefactBuild object
     """
+    store = artefact.store
+    assert store is not None, f"Store is not set for the artefact {artefact.id}"
+
     for build in artefact.builds:
         arch = build.architecture
         channel_map = get_channel_map_from_snapcraft(
             arch=arch,
-            snapstore=artefact.source["store"],
+            snapstore=store,
             snap_name=artefact.name,
         )
-        track = artefact.source.get("track", "latest")
+        track = artefact.track
```

Comment:
Technically if an Artefact is a snap then it should have a track. I've added a validation on `test-executions` endpoint to make sure that this is the case. While technically `artefact.source.get("track", "latest")` can be kept, I prefer to actually error out in this case just in case (also it messes up database uniqueness constraint if there is a null value).

Reply:
That makes sense, it would definitely be better to have an error so that we can know something in the database is not right. Thanks for clarifying it.